src/Button/index.js

```diff
@@ -0,0 +1,271 @@
+         textColor, textSize, textTransform, textWeight, onClick,
+         backgroundColor, border, borderRadius, color, fontFamily,
+         height, padding, minWidth, width,
+         hoverBackgroundColor, hoverBorder
+      } = this.props
+
+      /* eslint-disable object-property-newline */
+      const responsiveStyles = generateResponsiveStyles({
+         backgroundColor, border, borderRadius, color, fontFamily,
+         height, padding, minWidth, width
+      })
+
+      const hoverResponsiveStyles = generateResponsiveStyles({
+         backgroundColor: hoverBackgroundColor,
+         border: hoverBorder
+      })
+      /* eslint-enable */
+
+      const textStyle = {}
+      const buttonStyle = {
+         alignItems: 'center',
+         cursor: 'pointer',
+         display: 'inline-flex',
+         justifyContent: 'center',
+         outline: 'none',
+         position: 'relative',
+         transition: '200ms',
+         ...this.props.style,
+         ...responsiveStyles
+      }
+
+      // ====================================================
+      // Element
+      // ====================================================
+
+      // Abilty to use custom tag (div, button, span)
+      const ButtonElement = tagName
+      const conditionalButtonProps = {}
+
+      // If Button is used <button>, add type attribute
+      if (tagName === 'button') {
+         conditionalButtonProps.type = type
+      }
+
+      // ====================================================
+      // Mode
+      // ====================================================
+
+      if (mode === 'disabled') {
+         buttonStyle.opacity = '0.3'
+         buttonStyle.cursor = 'default'
+         conditionalButtonProps.disabled = true
+      } else if (mode === 'loading') {
+         textStyle.opacity = 0
+         buttonStyle.cursor = 'default'
+         conditionalButtonProps.disabled = true
+      } else if (mode === 'pulsing') {
+         buttonStyle.animation = `${pulsingAnimation} 2000ms infinite linear`
+      }
+
+      // Enable hover and active styles for normal and pulsing button modes
+      // No hover/active styles for modes: loading or disabled
+      if (mode === 'normal' || mode === 'pulsing') {
+         buttonStyle['&:hover'] = {
+            ...hoverResponsiveStyles
+         }
+         buttonStyle['&:active'] = {
+            transform: 'translateY(3px)'
+         }
+      }
+
+      return (
+         <ButtonElement
+            {...conditionalButtonProps}
+            css={buttonStyle}
+            className={className}
+            onClick={onClick}>
+            {children}
+            {name && (
+               <Text
+                  color={textColor}
+                  size={textSize}
+                  style={textStyle}
+                  transform={textTransform}
+                  weight={textWeight}>
+                  {name}
+               </Text>
+            )}
+            {mode === 'loading' && (
```

Comment:
Just thinking about the interaction between children (including `name`) when the `mode` is `loading`. I guess neither `name` nor `children` show up, but the width seems to be dictated by the children, which doesn't feel predictable. What is the intended behaviour here? I'd update the api options to reflect whatever the intention is.